09week/checkpoint3/script.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+// API Website https://blockchain.info/api/exchange_rates_api
+
+class BitcoinComponent extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			apiData: [],
+			selectedCurrency: ["Click a currency to see the current value in Bitcoin"]
+		}
+		this.displayData = this.displayData.bind(this);
+	}
+
+	componentWillMount(){
+		// Fetch Call
+		const url = `https://blockchain.info/ticker`; 
+			// show the url in the console
+			console.log("Fetch: ", URL); 
+			// define options
+			const options = {
+				method: 'GET',
+				headers: {'content-type': 'application/json'}
+			}
+		// Call data
+		fetch(url, options)
+		.then((response) => {
+			// show the response in the console
+            console.log("Response: ", response); 
+            // return the api data
+			return response.json();
+		})
+		// Store data
+		.then((data) => {		
+			// console log data
+            console.log("Data: ", data); 
+            // store data
+			this.setState({apiData:data})
+			// console log object state
+			// const objectLog = this.state.apiData;
+			// const propertyLog = this.state.apiData['USD']['buy'];
+			// console.log("New apiData State: ", objectLog);
+			// console.log("Property USD Buy: ", propertyLog);
+		});
+	}
+
+	// Display data on button click
+	displayData(event) {
+		if (Object.keys(this.state.apiData).length) {
```

Comment:
I would have liked to see this as a loop , or let me select from an option of currencies, instead of hardcoding them.